gatorgrade/output/output_functions.py

```diff
@@ -1,7 +1,49 @@
-"""Add docstring."""
+"""
+This module is used for storing the main functions requested.
 
+The requested functions are located at the Github Issue Tracker
+for the output team. For instance, functions dealing with percentage
+output, description output, and colorization of text.
+"""
 
-# This class is used for storing the main functions requested
-# from the Github Issue Tracker for the output team.
-# For instance, functions dealing with percentage output,
-# description output, and colorization of text.
+import gator
+
+
+def run_commands_and_return_results(commands_input):
+    """
+    Receive commands and send results to other output methods.
+
+    Commands are received as dictionary of two keys, shell commands / gator
+    commands.
+
+    An Example of input for this function is as follows :
+
+    {'shell': [{'description': 'Run program', 'command': 'mdl'}],
+    'gatorgrader': [['--description', 'do command', 'commandType',
+    '--arg', '1', '--directory', './home', '--file', 'file.py']]}
+    """
+    # pylint: disable=W0703
+    # Get first element in list, which is gatorgrader commands
+    gatorcommands = commands_input.get("gatorgrader")
+    results = []
+
+    for command in gatorcommands:
+        # If command is formatted incorrectly in yaml files,
+        # catch the exception that would be returned and print
+        try:
+            result = gator.grader(command)
+        except Exception as command_exception:

```

Comment:
I've brought this up before and I'd like to hear from you the reasoning behind this. Is there a reason you're catching all exceptions instead of specifying a type of exception here? I'm ok with catching anything as long as there is some justification for it. It would be great if we can discuss that before merging.

Reply:
We talked with Maria about this point and she said for now it would be better to create a catch-all exception handler rather than looking for specific exceptions.

Reply:
ok, that's fine! I was not aware that you've already agreed on that. Thanks for letting me know.